### ISSUE TITLE
Two Stage Training

### ISSUE THREAD

**Issue by cscyangj** [oawiles/X2Face#5]
How to run two stage training as described in paper?

**Comment by oawiles** [comment]
We haven't released this but may do it in the future.

**Comment by KeyKy** [comment]
@oawiles I want to ask some question about the The second stage training
1. Did you only use an identity loss? Did you use an adversarial loss?
2. How do you select driven frames in training? Did you select them from video with same identity to train the model?

**Comment by oawiles** [comment]
1. We didn't use an adversarial loss.
2. In the first stage: it's from the same video to use a photometric loss. In the second stage (in addition to the photometric loss), we use an additional loss. In this case, we use a different video but then only a content loss. Thus in the second stage for a given input image we would use two driving videos. One would be from the same video and have the photometric + content loss, the second would be from a different video and have the content loss. 

Please look at the paper for a more detailed explanation.

Hope this helps.

**Comment by mrgloom** [comment]
What is `photometric loss`? is it just L1 loss on pixels?

**Comment by nihaomiao** [comment]
Hi, we want to compare our model with X2Face in other datasets.  Could you release Two-stage training code so we can compare results fairly? @oawiles

**Comment by oawiles** [comment]
Hi,

Unfortunately, we do not plan on releasing that model. However, as stated
in the paper, there is a difference but it is not that large -- the large
failure cases are consistent between both training procedures. I think if
you have a clear improvement over the basic model then it should be fine to
say that it improves over X2Face.

Hope that helps,
Olivia

**Comment by oawiles** [comment]
We use a photometric loss to describe an 'image' based loss. It can be
either an L1 loss or an L1 + Perceptual loss in this case.
